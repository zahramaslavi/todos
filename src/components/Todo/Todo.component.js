import React from 'react';

import './Todo.component.scss';

const Todo = (props) => {

    return (
        <div className="todo">
            <div>
                <input 
                    type="checkbox"  
                    checked={props.todo.completed}
                    onChange={() => props.onToggleComplete()}
                />
                <span className={props.todo.completed ? 'completed' : 'active'}>{props.todo.description}</span>
                
            </div>
            <span className="delete" onClick={() => props.onDeleteTodo()}>x</span>
        </div>
    );
}

export default Todo;