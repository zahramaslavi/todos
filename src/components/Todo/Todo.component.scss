.todo {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding-top: .5em;
    padding-bottom: .5em;
    padding-left: 1em;
    padding-right: 1em;
    border-bottom: solid 1px lightgray;

    div input {
        margin-right: .5em;
        cursor: pointer;
    }

    div span.completed {
        text-decoration: line-through;
    }

    .delete {
        cursor: pointer;
        padding: .5em
    }
}