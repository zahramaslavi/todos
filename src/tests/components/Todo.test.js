import React from 'react';
import { shallow } from 'enzyme';
import Todo from '../../components/Todo/Todo.component';

test('should test Todo component', () => {
    const wrapper = shallow(<Todo todo={{description: 'test',completed: false, id: 1}} />);
    expect(wrapper).toMatchSnapshot();
});

test('should test Todo component deleteTodo', () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(
        <Todo 
            todo={{description: 'test',completed: false, id: 1}}
            onDeleteTodo={onSubmitSpy}
        />);
    
    const delBtn = wrapper.find('div.todo span.delete');
    
    delBtn.simulate('click', { bubbles: true });

    expect(onSubmitSpy).toHaveBeenCalled();
});

test('should test Todo component toggleComplete', () => {
    const onSubmitSpy = jest.fn();
    
    const wrapper = shallow(
        <Todo 
            todo={{description: 'test',completed: false, id: 1}}
            onToggleComplete={onSubmitSpy}
        />);

    const toggleCheckbox = wrapper.find('div.todo div input');
    
    toggleCheckbox.simulate('change');

    expect(onSubmitSpy).toHaveBeenCalled();
});