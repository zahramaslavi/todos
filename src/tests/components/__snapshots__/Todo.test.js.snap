// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Todo component 1`] = `
<div
  className="todo"
>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="active"
    >
      test
    </span>
  </div>
  <span
    className="delete"
    onClick={[Function]}
  >
    x
  </span>
</div>
`;
